### ISSUE TITLE
Bottle Generation Problems

### ISSUE THREAD

**Issue by DomT4** [Homebrew/homebrew-versions#791]
Opening this to track #785.

Problem:

```
==> Applying patch
Applying: syncthing010 0.10.30 (new formula)
==> Patch changed:
 syncthing010.rb | 56 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 56 insertions(+)
brew bottle --merge --write syncthing010-0.10.30.mavericks.bottle.rb syncthing010-0.10.30.mountain_lion.bottle.rb syncthing010-0.10.30.yosemite.bottle.rb --tap=Homebrew/versions
Error: Bottle block addition failed!
==> syncthing010
  bottle do
    root_url "https://homebrew.bintray.com/bottles-versions"
    cellar :any
    sha256 "8c87238e7718732fb27f1dbc3dd00747fb128629b8e273728761ad5202a3650a" => :yosemite
    sha256 "cae892790cf0744cfbe6417572e6dd53f8c429e1dd23ee6d5aae0040a76dec49" => :mavericks
    sha256 "10ef93e025ff25fe37c4c1ba81484678e7c13d9a61d9c9780af4a0b627acbc2a" => :mountain_lion
  end
Error: Failure while executing: brew bottle --merge --write syncthing010-0.10.30.mavericks.bottle.rb syncthing010-0.10.30.mountain_lion.bottle.rb syncthing010-0.10.30.yosemite.bottle.rb --tap=Homebrew/versions
Build step 'Execute shell' marked build as failure
```

Doesn't seem to a regex thing. Tests passed fine on testing the Syncthing010 bottle.

**Comment by DomT4** [comment]
Another one in #801.

```
==> Patch changed:
 docker150.rb | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)
brew bottle --merge --write docker150-1.5.0.mavericks.bottle.rb docker150-1.5.0.mountain_lion.bottle.rb docker150-1.5.0.yosemite.bottle.rb --tap=Homebrew/versions
Error: Bottle block addition failed!
==> docker150
  bottle do
    root_url "https://homebrew.bintray.com/bottles-versions"
    cellar :any
    sha256 "2256ee87c0119e6103017426be424fe90c7c5a6c04db41b74c8d92878063cb4a" => :yosemite
    sha256 "b324f9255ec8cd89955a149104953c8358ced91163c52caa798802ecb39e871c" => :mavericks
    sha256 "9f7aa93fee84ef44080e9b72e654b98189aae3c68731582d8d06cd9b305f07ee" => :mountain_lion
  end
Error: Failure while executing: brew bottle --merge --write docker150-1.5.0.mavericks.bottle.rb docker150-1.5.0.mountain_lion.bottle.rb docker150-1.5.0.yosemite.bottle.rb --tap=Homebrew/versions
Build step 'Execute shell' marked build as failure

Deleting project workspace... 
done

Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in  1 second
Finished: FAILURE
```

Any thoughts @Homebrew/owners? This started some time in the last week. Before that, no issues at all. I've checked the produced bottle regexes with ` brew tests ` and nothing has been flagged as failing.

**Comment by mikemcquaid** [comment]
No idea, can you reproduce locally or get a backtrace there with `--debug`?

**Comment by DomT4** [comment]
Yes to repro. Nothing useful coughed up on the `--debug` though:

```
Dominyks-MacBook-Pro ~> brew bottle --merge --write --debug docker150-1.5.0.yosemite.bottle.rb --tap=Homebrew/versions
==> docker150
  bottle do
    root_url "https://homebrew.bintray.com/bottles-versions"
    cellar :any
    sha256 "b5abca67f1534c127a1323e40e7efb4eb6762b49fe71fa623ecfaccd6527bd95" => :yosemite
  end
/usr/local/Library/brew.rb (Formulary::TapLoader): loading /usr/local/Library/Taps/homebrew/homebrew-versions/docker150.rb
Error: Bottle block addition failed!
```

**Comment by xu-cheng** [comment]
Looks like a regex error in [bottle addition](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/cmd/bottle.rb#L260). It failed to consider `url` can be multiple lines.

**Comment by mikemcquaid** [comment]
Yeh, that code is never going to work for 100% of cases. When it doesn't I you can add an empty `bottle do` to the formula.

**Comment by xu-cheng** [comment]
I have work out for a possible fix in https://github.com/Homebrew/homebrew/pull/39137

**Comment by DomT4** [comment]
> When it doesn't I you can add an empty bottle do to the formula.

Nice. I wondered about just pasting the bottle block in if it refused to do it automatically, but didn't know if that'd screw anything else up.

> I have work out for a possible fix in Homebrew/homebrew#39137

Thanks Xu! Looks great.
